channels/rdpgfx/client/rdpgfx_main.c

```diff
@@ -689,22 +953,31 @@ static UINT rdpgfx_recv_cache_import_reply_pdu(RDPGFX_CHANNEL_CALLBACK* callback
 	if (!Stream_CheckAndLogRequiredLength(TAG, s, 2ull * pdu.importedEntriesCount))
 		return ERROR_INVALID_DATA;
 
-	pdu.cacheSlots = (UINT16*)calloc(pdu.importedEntriesCount, sizeof(UINT16));
+	if (pdu.importedEntriesCount > RDPGFX_CACHE_ENTRY_MAX_COUNT)
+		return ERROR_INVALID_DATA;
 
-	if (!pdu.cacheSlots)
+	if (Stream_GetRemainingLength(s) / 2 < (size_t)pdu.importedEntriesCount)
```

Comment:
Better use `Stream_CheckAndLogRequiredLength` so we are notified about protocol violations in log messages

Reply:
hum... there's already such a check a few lines above, should we just remove this check then?

Reply:
oh, yes, true.
looks like this is a duplicate, indeed.